### PR TITLE
[JSE-01] Add 'list-podcast' and 'search-input' components

### PR DESCRIPTION
**Pull request by josepenagosGL**
This is the main structure for the list and search (scoped with the list) components.

### CODE REVIEW

File: src/components/list-podcast/list-podcast.vue [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      <v-spacer></v-spacer>
+
+      <search-input :keypressEvent="'search-track'" @search-track="searchTrack" />
+    </v-toolbar>
+
+    <v-list two-line flat>
+      <v-list-item-group v-model="selected">
+        <template v-for="(item, index) in list">
+          <v-list-item :key="item.title">
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/list-podcast.vue]: Could we guarantee that **title** key is unique value? For this case I recommend using **trackId** key

File: src/components/list-podcast/list-podcast.vue [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<template>
+  <v-card max-width="500" class="mx-auto">
+    <v-toolbar color="green" dark>
+      <v-toolbar-title>{{ listTitle }}</v-toolbar-title>
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/list-podcast.vue]: **listTitle** is redundant due the component's context name, I suggest using just **title**

File: src/components/list-podcast/list-podcast.vue [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      <search-input :keypressEvent="'search-track'" @search-track="searchTrack" />
+    </v-toolbar>
+
+    <v-list two-line flat>
+      <v-list-item-group v-model="selected">
+        <template v-for="(item, index) in list">
+          <v-list-item :key="item.title">
+            <v-list-item-action>
+              <v-list-item-avatar>
+                <v-img :src="item.artworkUrl100"></v-img>
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/list-podcast.vue]: 1. Do we need this resolution? I suggest using smaller if it possible
2. Please use `alt` attr
3. Please remove `></v-img>` just `/>`

File: src/components/list-podcast/list-podcast.vue [hunk 4]
```diff
@@ -0,0 +1,46 @@
+<template>
+  <v-card max-width="500" class="mx-auto">
+    <v-toolbar color="green" dark>
+      <v-toolbar-title>{{ listTitle }}</v-toolbar-title>
+
+      <v-spacer></v-spacer>
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/list-podcast.vue]: Is it possible to use auto-close tags here? something like `<v-spacer />`

File: src/components/list-podcast/list-podcast.vue [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        <template v-for="(item, index) in list">
+          <v-list-item :key="item.title">
+            <v-list-item-action>
+              <v-list-item-avatar>
+                <v-img :src="item.artworkUrl100"></v-img>
+              </v-list-item-avatar>
+            </v-list-item-action>
+
+            <v-list-item-content>
+              <v-list-item-title v-text="item.trackName"></v-list-item-title>
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/list-podcast.vue]: SAB: Auto close tags

File: src/components/list-podcast/list-podcast.vue [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+              <v-list-item-subtitle v-html="getSubTitle(item.artistName, item.primaryGenreName)" />
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/list-podcast.vue]: 1. Please use computed prop here
2. Do we need to use html directive?

File: src/components/list-podcast/list-podcast.vue [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            </v-list-item-content>
+
+            <v-list-item-icon>
+              <v-icon
+                @click="$emit('add-favorite', item.trackId)"
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/list-podcast.vue]: I suggest moving this into a method, I think is more readable.

File: src/components/list-podcast/list-podcast.vue [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+                :color="item.active ? 'deep-purple accent-4' : 'grey'"
+              >mdi-star-outline</v-icon>
+              <v-spacer></v-spacer>
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/list-podcast.vue]: SAB

File: src/components/list-podcast/list-podcast.vue [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+              <v-icon
+                @click="$emit('play-track', item.trackId)"
+                :color="item.active ? 'deep-purple accent-4' : 'grey'"
+              >mdi-play-circle-outline</v-icon>
+            </v-list-item-icon>
+          </v-list-item>
+
+          <v-divider v-if="index + 1 < list.length" :key="index"></v-divider>
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/list-podcast.vue]: Using **index** as key is an anti-pattern, please use **trackId**

File: src/components/list-podcast/script.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+export default {
+  name: 'ListPodcast',
+  data() {
+    return {
+      selected: false,
+    };
+  },
+  methods: {
+    getSubTitle(artistName, genreName) {
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/script.js:12]: Use **subtitle** computed prop

File: src/components/list-podcast/script.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+      type: Array,
+      default: () => [],
+    },
+    listTitle: {
+      type: String,
+      default: 'All Podcasts',
+    },
+  },
+  components: {
+    'search-input': SearchInput,
```
Comment (alejandrobenjumeaGL) [inline src/components/list-podcast/script.js]: Please use only shorthand: **SearchInput**

File: src/components/search-input/script.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
+export default {
+  name: 'SearchInput',
+  methods: {
+    change(track) {
+      if (track) {
+        this.$emit(this.changeEvent, track);
```
Comment (alejandrobenjumeaGL) [inline src/components/search-input/script.js]: what about using trim()?

File: src/components/search-input/script.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+      }
+    },
+    keypress(e) {
```
Comment (alejandrobenjumeaGL) [inline src/components/search-input/script.js]: Please avoid using parameter abbreviation, please use expressive naming, e.g **event**

File: src/components/search-input/script.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+      this.$emit(this.keypressEvent, e.srcElement.value);
+    },
+  },
+  props: {
+    changeEvent: {
+      type: String,
```
Comment (alejandrobenjumeaGL) [inline src/components/search-input/script.js:22]: q: type shouldn't be function?

File: src/components/search-input/script.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+      default: '',
+    },
+    keypressEvent: {
+      type: String,
```
Comment (alejandrobenjumeaGL) [inline src/components/search-input/script.js:26]: q: type shouldn't be function?

File: src/components/search-input/search-input.vue [hunk 16]
```diff
@@ -0,0 +1,17 @@
+<template>
+  <v-text-field
+    label="Search.."
```
Comment (alejandrobenjumeaGL) [inline src/components/search-input/search-input.vue]: 1. Did you mean ...?
2. `label` could be a prop with `Search...` as default value

File: src/components/search-input/search-input.vue [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+    append-icon="mdi-magnify"
+    @change="change"
+    @keypress="keypress"
+  />
+</template>
+
+<script src="./script.js"></script>
+
+<style scoped>
+.v-input {
```
Comment (alejandrobenjumeaGL) [inline src/components/search-input/search-input.vue]: I suggest being consistent, please move this code in separated file

**Review by alejandrobenjumeaGL — CHANGES_REQUESTED** [review]

File: src/components/list-podcast/list-podcast.vue (hunk 6 above)
Comment (josepenagosGL) [reply src/components/list-podcast/list-podcast.vue]: I think it's not possible to pass parameters to computed properties unless we create a setter function that previously assigns the values that eventually will be used for the computed property.

Something like this:

https://ednsquare.com/question/how-to-pass-parameters-in-computed-properties-in-vue-js-------MQVlHT

I think it's too much. The other option will be printing the value directly v-text (that I 've already changed for v-text.

File: src/components/list-podcast/list-podcast.vue (hunk 9 above)
Comment (josepenagosGL) [reply src/components/list-podcast/list-podcast.vue]: trackId has been already used for a sibling tag, the only id we can use here is this unless we found another unique identifier in our data source.

File: src/components/list-podcast/script.js (hunk 10 above)
Comment (josepenagosGL) [reply src/components/list-podcast/script.js:12]: Same as the other method suggested to be changed to a computed property

File: src/components/list-podcast/script.js (hunk 11 above)
Comment (josepenagosGL) [reply src/components/list-podcast/script.js]: Good catch, I thought it was necessary to use dash case even when including the component to call it within the template.

File: src/components/search-input/script.js (hunk 12 above)
Comment (josepenagosGL) [reply src/components/search-input/script.js]: Good suggestion, created a function to parse clean the search term and then pass it.

File: src/components/search-input/script.js (hunk 14 above)
Comment (josepenagosGL) [reply src/components/search-input/script.js:22]: No, remember consumers should just pass the event we're going to emit whether on change or keypress/keydown. Do you think we should change the parameter's name instead?

File: src/components/list-podcast/list-podcast.vue (hunk 9 above)
Comment (alejandrobenjumeaGL) [reply src/components/list-podcast/list-podcast.vue]: We can prefix it with something else ``${prefix}${tackId}``

**Pull request merged**